helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -628,18 +652,44 @@ private static PipelineRegistry createTaskRegistry(String pipelineName) {
     }
   }
 
+  private static PipelineRegistry createManagementModeRegistry(String pipelineName) {
+    logger.info("Creating management mode registry");
+    synchronized (GenericHelixController.class) {
+      // cluster data cache refresh
+      Pipeline dataRefresh = new Pipeline(pipelineName);
+      dataRefresh.addStage(new ReadClusterDataStage());
+
+      // cluster management mode process
+      Pipeline managementMode = new Pipeline(pipelineName);
+      managementMode.addStage(new ManagementModeStage());
+
+      PipelineRegistry registry = new PipelineRegistry();
+      Arrays.asList(
```

Comment:
where is onController change?

Reply:
+1 Don't we need onController change event?

Reply:
We'll set `_runManagementModePipeline=true` and `OnDemandRebalance` for the onController change event. No need to add an additional event type.

Reply:
Wait. If the cluster in pause mode, it will run the management pipline right? But if there is a controller change, you will miss that event.